File: javascript/ankit/PascalTriangle.js
// link to problem https://leetcode.com/problems/pascals-triangle/

var generate = function (noRows) {
  let arr = [[1], [1, 1]];
  if (noRows == 1) return [[1]];
  if (noRows == 2) return arr;
  for (let i = 3; i <= noRows; i++) {
    let temp = [1];
    for (let j = 2; j < i; j++) {
      temp.push(arr[i - 2][j - 2] + arr[i - 2][j - 1]);
    }
    temp.push(1);
    arr.push(temp);
  }
  return arr;
};

File: java/ankit_kanyal/arrays/PascalTriangle.java
// link- https://workat.tech/problem-solving/practice/pascals-triangle

class Solution {
	
	int[] pascalTriangleRow(int rowNo) {
		int [] curRow=new int[rowNo];
		curRow[0]=1;
		if(rowNo==1){
			return new int[]{1};
		}
		int [] preRow=pascalTriangleRow(rowNo-1);
		for(int i=1;i<rowNo-1;i++){
			curRow[i]=preRow[i]+preRow[i-1];
		}
		
		curRow[rowNo-1]=1;
		return curRow;
	}
}